pkg/config/transform.go

```diff
@@ -69,8 +69,11 @@ func TransformSimpleToClusterConfig(ctx context.Context, runtime runtimes.Runtim
 		simpleConfig.ExposeAPI.HostIP = k3d.DefaultAPIHost
 	}
 
-	kubeAPIExposureOpts := &k3d.ExposureOpts{
-		Host: simpleConfig.ExposeAPI.Host,
+	kubeAPIExposureOpts := &k3d.ExposureOpts{}
+	if simpleConfig.ExposeAPI.Host == k3d.DefaultAPIHost {
+		kubeAPIExposureOpts.Host = simpleConfig.ExposeAPI.HostIP
+	} else {
+		kubeAPIExposureOpts.Host = simpleConfig.ExposeAPI.Host
```

Comment:
I think this should move a few lines up, as it would break expected functionality, if someone specifies
```yaml
kubeAPI:
  host: "0.0.0.0"
  hostIP: "1.2.3.4"
```
right?

We could do something like
```Go
if simpleConfig.ExposeAPI.HostIP == "" {
	simpleConfig.ExposeAPI.HostIP = k3d.DefaultAPIHost
}
if simpleConfig.ExposeAPI.Host == "" {
	simpleConfig.ExposeAPI.Host = simpleConfig.ExposeAPI.HostIP
}
```
starting at line 65

Reply:
Hello @iwilltry42 

Just wanted to clarify what the kubecfg address should be in the various use cases combinations of Host and HostIP, because I indeed did not consider the usage of both...

| UseCase | Host | IP | Expected KubCfg Server Address |
| -- | -- | -- | --  | 
| uc1| - | - | 0.0.0.0 | 
| uc2| X | - | Host |
| uc3| - | X | IP | 
| uc4|X | X | IP | 

In the case you mention, (Host & HostIP), is that correct to assume HostIP should be retained ? I guess in both my implementation and with the suggestion we have an issue (host is chosen).

Attaching the clusters config usecases
[usecases.zip](https://github.com/rancher/k3d/files/6092011/usecases.zip)

Reply:
Hey @fabricev , I think it should be

| UseCase | Host | HostIP | Expected KubCfg Server Address |
| -- | -- | -- | --  | 
| uc1| - | - | 0.0.0.0 | 
| uc2| X | - | Host |
| uc3| - | X | HostIP | 
| uc4|X | X | Host | 

(just changed uc4 to use the Host)

Reply:
Hello @iwilltry42, that clarifies, thanks! Corrected as suggested (and tested the various usecases above).